it/test/connectors/PropertyTaskListDataConnectorISpec.scala

```diff
@@ -0,0 +1,169 @@
+import models.tasklist.TaskTitle.{ESA, ForeignProperty, JSA, UkForeignProperty, UkProperty}
+import models.tasklist.{SectionTitle, TaskListSection, TaskListSectionItem, TaskStatus}
+import models.{APIErrorBodyModel, APIErrorModel, APIErrorsBodyModel}
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import play.api.http.Status._
+import play.api.libs.json.Json
+import uk.gov.hmrc.http.client.HttpClientV2
+import uk.gov.hmrc.http.{HeaderCarrier, HeaderNames, SessionId}
+import utils.{ConnectorIntegrationTest, MockAppConfig}
+
+import java.time.{LocalDate, ZoneId}
+import scala.concurrent.Await
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration.Duration
+
+class PropertyTaskListDataConnectorISpec extends ConnectorIntegrationTest {
+  private val taxYear = LocalDate.now(ZoneId.systemDefault()).getYear
+  private val mtdItIdHeader = ("mtditid", "1234567890")
+  private val requestHeaders = Seq(new HttpHeader("mtditid", "1234567890"))
+  val nino :String = "123456789"
+
+  private val underTest: PropertyTaskListDataConnector = new PropertyTaskListDataConnector(httpClientV2, new MockAppConfig())
+  def propertyTaskListDataUrl: String = s"/income-tax-property/$taxYear/tasks/$nino"
+
+  "PropertyTaskListDataConnector" should {
+
+    val expectedResult: Seq[TaskListSection] =
+      Seq(
+        TaskListSection(
+          sectionTitle = SectionTitle.UkPropertyTitle,
+          taskItems = Some(List[TaskListSectionItem](
+            TaskListSectionItem(UkProperty, status = TaskStatus.NotStarted, Some("url"))
+          ))
+        ),
+        TaskListSection(
+          sectionTitle = SectionTitle.ForeignPropertyTitle,
+          taskItems = Some(List[TaskListSectionItem](
+            TaskListSectionItem(ForeignProperty, status = TaskStatus.NotStarted, Some("url"))
+          ))
+        ),
+        TaskListSection(
+          sectionTitle = SectionTitle.UkForeignPropertyTitle,
+          taskItems = Some(List[TaskListSectionItem](
+            TaskListSectionItem(UkForeignProperty, status = TaskStatus.Completed, Some("url"))
+          ))
+        )
+      )
+
+    val responseBody = Json.toJson(expectedResult).toString()
+    implicit val hc: HeaderCarrier = HeaderCarrier(sessionId = Some(SessionId("sessionIdValue"))).withExtraHeaders(mtdItIdHeader)
+
+    "include internal headers" when {
+      val headersSentToPropertyTaskList = Seq(new HttpHeader(HeaderNames.xSessionId, "sessionIdValue"))
+
+      "the host is 'Internal'" in {
+        stubGetWithResponseBody(propertyTaskListDataUrl, OK, responseBody, headersSentToPropertyTaskList)
+
+        Await.result(underTest.get(taxYear,nino), Duration.Inf) shouldBe Right(Some(expectedResult))
+      }
+
+      "the host is 'External'" in {
+        stubGetWithResponseBody(propertyTaskListDataUrl, OK, responseBody, headersSentToPropertyTaskList)
+
+        Await.result(underTest.get(taxYear,nino), Duration.Inf) shouldBe Right(Some(expectedResult))
```

Comment:
For testing the async functions, I think we should either be using the `await` helper from ScalaTest or the postfix helper `.futureValue`